### PR TITLE
operator expressions: add &raw

### PR DESCRIPTION
**Pull request by RalfJung**
Accompanying PR for https://github.com/rust-lang/rust/pull/127679.
Fixes https://github.com/rust-lang/rust/issues/64490.

I reworded things to avoid the "address-of" terminology, since a pointer is not just an address.

### CODE REVIEW

File: src/expressions/operator-expr.md
Comment (RalfJung) [inline src/expressions/operator-expr.md:104]: I assume this will fail in CI since the operator is not stable yet. Not sure how you usually stage this.

File: src/expressions/operator-expr.md
Comment (ehuss) [inline src/expressions/operator-expr.md:45]: Looks to be missing a back tick.

```suggestion
> &nbsp;&nbsp; | (`&`|`&&`) `raw` `const` [_Expression_]\
```

File: src/expressions/operator-expr.md
Comment (ehuss) [inline src/expressions/operator-expr.md:87]: What `expr` is hasn't been specified (as the old text it was referring to has been removed).

```suggestion
The operand expression of the raw pointer operator is evaluated in place expression context.
```

**Review by ehuss — COMMENTED** [review]
Thanks!

Should the last sentence of the section https://github.com/rust-lang/reference/blob/master/src/types/pointer.md#raw-pointers-const-and-mut be updated to use the new syntax instead of addr_of?

And in general, should all other uses of addr_of in the reference be changed to use the new syntax? My understanding is that is now the preferred style, is that correct?

File: src/expressions/operator-expr.md
Comment (ehuss) [reply src/expressions/operator-expr.md:104]: Yea, CI will just be red until the stabilization PR hits nightly, after which point we can merge this.

File: src/expressions/operator-expr.md
Comment (RalfJung) [inline src/expressions/operator-expr.md:84]: I am not sure what is the best thing to call it. Another possible name here might be "Raw borrow expression/operator", to make it clear that this is very similar to reference-creating operations. After all this is in a section titled "borrow operators".

Comment (traviscross) [reply src/expressions/operator-expr.md:84]: "Raw borrow" sounds right to me.  A "raw borrow" creates a "raw reference", which, yes, is also a "raw pointer".

Comment (RalfJung) [reply src/expressions/operator-expr.md]: All right, I have switched the PR to use "raw borrow operator".

File: src/expressions.md
Comment (traviscross) [inline src/expressions.md:279]: If you would, just make the anchor "raw borrow" and correspondingly simplify the use of it above, and move this line to its alphabetized spot (even though it muddies up the diff).

**Review by traviscross — APPROVED** [review]
Otherwise this looks good to me.  We'll merge this once the stabilization PR is merged.

**Pull request merged**
